### PR TITLE
Cypress: set up locally and in CI, write first test

### LINKED ISSUE #8: Set up testing
(Finish?) setting up testing frameworks

### PR DESCRIPTION
**Pull request by BCerki**
This PR  addresses issue: #107 and #101

Includes tests? y
Rebased on main address to conflicts? n/a
Updated docs? y

Proposed changes:
-write a Cypress test that checks if a gov user can create and save a draft of a CWU opportunity. This involved the following pre-reqs:
-- add an endpoint that authenticates a user for testing (endpoint only exists in dev)
--update docker-compose to use latest postgres
-- create two fixtures
-- update docs with instructions for running the cypress tests
-- tweak some configs
--create a github action file to run the tests in CI

Additional notes
-there's some good conversation in this PR: https://github.com/button-inc/digital_marketplace/pull/111

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ module.exports = {
       "@typescript-eslint/no-var-requires": "off",
       "@typescript-eslint/no-empty-function": "off",
       "react/no-children-prop": "off",
+      "@typescript-eslint/ban-ts-comment": "warn"
```
Comment (BCerki) [inline .eslintrc.js:35]: Sorry... had to introduce some tech debt to get the test endpoint set up

File: README.md [hunk 2]
```diff
@@ -156,36 +156,37 @@ It is recommended that developers use the following scripts defined in `package.
...
+| Script Name                             | Description                                                                                                                                                                                                        |
+| --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `cypress:run`                          | Runs the Cypress tests. (You must first manually start the app for the tests to have something to run against.) NOTE: The test set up and clean up will wipe and recreate the local database.                      |
+| `cypress:open`                          | Opens the interactive Cypress test runner. (You must first manually start the app for the tests to have something to run against.) NOTE: The test set up and clean up will wipe and recreate the local database.   |
```
Comment (BCerki) [inline README.md:162]: These are the only scripts I changed; the other changes are just making the table wider

File: cypress/support/commands.js [hunk 3]
```diff
@@ -23,3 +23,10 @@
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+
+import 'cypress-file-upload';
+import '@testing-library/cypress/add-commands';
+
+Cypress.Commands.add('sqlFixture',(sqlFilename)=>{
+    cy.exec(`docker exec dm_db psql -U digitalmarketplace digitalmarketplace -f /workspace/cypress/fixtures/${sqlFilename}`) //need /workspace/ because that's the file structure in the db container
+})
```
Comment (BCerki) [inline cypress/support/commands.js:32]: This based on how [CAS does it-](https://github.com/bcgov/cas-cif/pull/194/files)-couldn't get theirs to work so tweaked, hopefully this is cool too

File: docker-compose.yml [hunk 4]
```diff
@@ -1,31 +1,34 @@
...
+  #   build:
+  #     context: ././
+  #     dockerfile: src/front-end/Dockerfile
+  #   ports:
+  #     - "3000:3000"
+  #   environment:
+  #     POSTGRES_URL: postgresql://digitalmarketplace:digitalmarketplace@db:5432/digitalmarketplace
+  #     NODE_ENV: ${NODE_ENV}
+  #   depends_on:
+  #     - db
```
Comment (BCerki) [inline docker-compose.yml:15]: I think I'd ultimately like to delete this--I don't want to run the app in a container when I'm developing locally; I want to run it with watch

Comment (mikevespi) [reply docker-compose.yml:15]: Local dev is the only reason this exists in the first place right?

**Review by mikevespi — APPROVED** [review]
This is great! 🚢

File: docker-compose.yml (hunk 4 above)
Comment (BCerki) [reply docker-compose.yml:15]: Yes

**Pull request merged**